app/controllers/steps_controller.rb

```diff
@@ -44,4 +46,14 @@ def next_path(params = nil)
   def step_navigation
     @step_navigation ||= StepNavigation.new(self)
   end
+
+  def maybe_skip
```

Comment:
If `skip?` is set to false, isn't this always going to end up with the same behavior? If that's the case I would imagine the `skip?` method will end up, in the future, being changed to result in a nondeterministic bool.

Could this work be deferred until that's needed?